frontend/src/stores/levels/actions.ts

```diff
@@ -16,4 +17,14 @@ async function addLevel(level: Level, role: Role): Promise<void> {
   role.levels?.push(level)
 }
 
-export { removeLevel, addLevel }
+async function editLevel(level: Level): Promise<void> {
+  await put<Level[]>(`/levels/${level.id}`, level)
+  levels.value = levels.value.map((item) => {
```

Comment:
You should update the levels for a specific role. That means, get the role of that level, and then update the levels array for that role